iOS Client/Routing/Routing.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  Routing.swift
+//  iOS Client
+//
+//  Created by Jonathan Witten on 10/4/17.
+//  Copyright © 2017 Mockingbird. All rights reserved.
+//
+
+import Foundation
+import ReSwift
+
+enum Route: Equatable {
+    
+    case base
+    case detail(item: Detailable)
+    
+    static func == (lhs: Route, rhs: Route) -> Bool {
+        if case .base = lhs, case .base = rhs {
+            return true
+        } else if case .detail = lhs, case .detail = rhs {
+            return true
+        } else {
+            return false
+        }
+    }
+ 
+    
+    static func viewController(forRoute route: Route, inTab tab: Tab) -> UIViewController? {
+        switch tab {
+        case .articles:
+            if case .base = route {
+                if let vc = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: "ArticlesController") as? MBArticlesViewController {
+                    return vc
+                }
+            }
+        case .bookmarks:
+            if case .base = route {
+                if let vc = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: "BookmarksController") as? MBBookmarksViewController {
+                    return vc
+                }
+            }
+        }
+        return nil
+    }
+}
+
+enum Tab: String {
```

Comment:
any reason we need `String` here? If we don't, we could do `enum Tab: Int` and use `rawValue` instead of needing the computed index property below.